allenact/algorithms/onpolicy_sync/runner.py

```diff
@@ -630,50 +645,94 @@ def experiment_name(self):
     def checkpoint_dir(
         self, start_time_str: Optional[str] = None, create_if_none: bool = True
     ):
-        folder = os.path.join(
-            self.output_dir,
-            "checkpoints",
-            self.config.tag()
-            if self.extra_tag == ""
-            else os.path.join(self.config.tag(), self.extra_tag),
-            start_time_str or self.local_start_time_str,
-        )
+        if self.logging_in_one_folder:
+            folder = os.path.join(
+                self.output_dir,
+                self.config.tag()
+                if self.extra_tag == ""
+                else os.path.join(self.config.tag(), self.extra_tag),
+                start_time_str or self.local_start_time_str,
+                "checkpoints",
+            )
+        else:
+            folder = os.path.join(
+                self.output_dir,
+                "checkpoints",
+                self.config.tag()
+                if self.extra_tag == ""
+                else os.path.join(self.config.tag(), self.extra_tag),
+                start_time_str or self.local_start_time_str,
+            )
         if create_if_none:
             os.makedirs(folder, exist_ok=True)
         return folder
 
     def log_writer_path(self, start_time_str: str) -> str:
-        path = os.path.join(
-            self.output_dir,
-            "tb",
-            self.config.tag()
-            if self.extra_tag == ""
-            else os.path.join(self.config.tag(), self.extra_tag),
-            start_time_str,
-        )
-        if self.mode == TEST_MODE_STR:
-            path = os.path.join(path, "test", self.local_start_time_str)
-        return path
-
-    def metric_path(self, start_time_str: str) -> str:
-        return os.path.join(
-            self.output_dir,
-            "metrics",
-            self.config.tag()
-            if self.extra_tag == ""
-            else os.path.join(self.config.tag(), self.extra_tag),
-            start_time_str,
-        )
+        if self.logging_in_one_folder:
+            if self.mode == TEST_MODE_STR:
+                checkpoint_parts = start_time_str
+                return os.path.join(
+                    *checkpoint_parts,
+                    "test",
+                    self.config.tag(),
+                    self.local_start_time_str,
+                )
```

Comment:
2. This is what I intend to. I check the `self.output_dir` is something like `'experiment_output'`, the root experiment dir. But I'd like to store testing results into the logging dir that contains the checkpoint. Yes, this assumes the checkpoint should follow some format `**/time_str/checkpoints/*.pt` and save the results in `**/time_str/test/`. I think one way is to provide an argument to let the user to specify the storing folder. Or we just let `wget <url> **/time_str/checkpoints/*.pt` to ensure the format?

Reply:
2. The problem is from user side. If the checkpoint is generated by their training script, then it can be quite nested and long str. Then the user has to specify both `-c` and `-o`. I am thinking about a way that if the user does not specify `-o` then just use `-c`'s folder. But how to check if the user specifies `-o`? Default value may be exactly what the user specifies..

Reply:
Good point. Perhaps we could have an `--infer_output_dir` parameter which will infer the output dir from the checkpoint path if the default value isn't specified?

Reply:
Good idea!